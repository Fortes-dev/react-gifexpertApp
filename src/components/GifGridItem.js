import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    return (

        //En vez de class, se usa className en react
        <div className="card animate__animated animate__backInLeft">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
