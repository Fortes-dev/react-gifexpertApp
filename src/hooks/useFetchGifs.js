import { useEffect, useState } from "react"
import { apiRequest } from "../utils/getFetchGifs";


export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect( () => {

        apiRequest(category).then(imgs => {
            setTimeout( () => {
                setState({
                    data: imgs,
                    loading: false,
                });
            }, 800);
            
        });

    }, []);


    return state; //{data:[], loading:true}

}