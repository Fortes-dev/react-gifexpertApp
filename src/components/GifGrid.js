import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category); 

    return (
        <> 
        <h3 className="animate__animated animate__pulse">{category}</h3>
        { loading && <p className="animate__animated animate__fadeOut">Loading...</p>}
        <div className="card-grid">
                {
                    data.map( img => 
                        (
                            <GifGridItem 
                                {...img}
                                key={img.id}/>
                        )
                    )
                }
           
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
