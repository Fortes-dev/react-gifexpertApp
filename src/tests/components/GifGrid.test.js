import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test suite de GifGrid', () => {
    
    const category = 'Berserk';

    test('Debe mostrar el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imagenes con nuestro useFetchGifs (custom hook)', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);

    })

});