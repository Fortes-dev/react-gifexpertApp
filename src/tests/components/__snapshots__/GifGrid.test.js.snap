// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite de GifGrid Debe mostrar el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    Berserk
  </h3>
  <p
    className="animate__animated animate__fadeOut"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
